Philips.CodeAnalysis.Common/GeneratedCodeDetector.cs

```diff
@@ -0,0 +1,88 @@
+﻿// © 2019 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Philips.CodeAnalysis.Common
+{
+	public class GeneratedCodeDetector
+	{
+		private readonly string _attributeName;
+		private readonly string _fullAttributeName;
+
+		public GeneratedCodeDetector()
+			: this(@"GeneratedCode", @"System.CodeDom.Compiler.GeneratedCodeAttribute")
+		{ }
+
+		public GeneratedCodeDetector(string attributeName, string fullAttributeName)
```

Comment:
Do you need this constructor overload, as it seems to be able to defy the purpose of this class. Therefore, consider removing it.

Reply:
Good question. It was there so that an Analyzer's Test (ie AvoidRedundantSwtichStatementAnalyzer) could dependency inject different attributes in order to bypass the CodeAnalysisFlags.Analyze behavior. But you're right - it's probably unnecessary now that GeneratedCodeDetector has its own tests. I'll remove it as part of the next step of changing AvoidRedundantSwtichStatementAnalyzer to use this.